vllm/v1/core/kv_cache_manager.py

```diff
@@ -200,6 +200,18 @@ def allocate_slots(
             raise ValueError(
                 f"num_tokens must be greater than 0, got {num_tokens}")
 
+        # If a computed block of a request is an eviction candidate (in the
+        # free queue and ref_cnt == 0), it cannot be counted as a free block
+        # when allocating this request.
+        num_evictable_computed_blocks = len(
+            [blk for blk in computed_blocks if blk.ref_cnt == 0])
```

Comment:
I'm a bit worry about the overhead of iterating computed_blocks especially for the case of long context prompt with long shared prefix, but I don't have a better solution now. I'll think about it over the weekend.